Assets/Scripts/Map/Partition.cs

```diff
@@ -0,0 +1,119 @@
+// or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+// Public License for more details.
+// 
+// You should have received a copy of the GNU General Public License along
+// with MAEPS. If not, see http://www.gnu.org/licenses/.
+//
+// Contributors 2025: 
+// Casper Nyvang Sørensen,
+// Christian Ziegler Sejersen,
+// Jakob Meyer Olsen
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+using Maes.Utilities;
+
+using UnityEngine;
+
+
+namespace Maes.Map
+{
+    public class Partition
+    {
+        public List<Vertex> Vertices { get; }
+        public Bitmap CommunicationZone { get; }
+        public int PartitionId { get; }
+        public Dictionary<(int, int), float> CommunicationRatio { get; }
+        public Dictionary<(int, int), Bitmap> IntersectionZones { get; }
+        public Dictionary<Vector2Int, Bitmap> CommunicationZones { get; }
+
+        public Partition(int partitionId, List<Vertex> vertices, Dictionary<Vector2Int, Bitmap> communicationZones)
+        {
+            PartitionId = partitionId;
+            Vertices = vertices;
+            CommunicationZones = communicationZones;
+            CommunicationZone = CreatePartitionCommunicationZone(communicationZones);
+            IntersectionZones = new Dictionary<(int, int), Bitmap>();
+            CommunicationRatio = new Dictionary<(int, int), float>();
+        }
+
+        public void CalculateIntersectionAndRatio(Partition otherPartition)
+        {
+            var key = (PartitionId, otherPartition.PartitionId);
+
+            if (AlreadyCalculated(key))
+            {
+                return;
+            }
+            var otherPartitionCommuncationZoneSubset = new Bitmap(0, 0, otherPartition.CommunicationZone.Width, otherPartition.CommunicationZone.Height);
+            foreach (var otherPartitionVertex in otherPartition.Vertices)
+            {
+                if (!otherPartition.CommunicationZones.TryGetValue(otherPartitionVertex.Position, out var zoneBitmap))
+                {
+                    throw new ArgumentException($"No communication zone found for vertex at position {otherPartitionVertex.Position}");
+                }
+                foreach (var currentPartitionVertex in Vertices)
+                {
+                    if (zoneBitmap.Contains(currentPartitionVertex.Position))
+                    {
+                        otherPartitionCommuncationZoneSubset.Union(zoneBitmap);
+                    }
+                }
+            }
+
+            var intersection = Bitmap.Intersection(CommunicationZone, otherPartitionCommuncationZoneSubset);
+            IntersectionZones[key] = intersection;
+
+            var ratio = CalculateRatio(intersection.Count, otherPartition.CommunicationZone.Count);
+            CommunicationRatio[key] = ratio;
+        }
+
+        private Bitmap CreatePartitionCommunicationZone(Dictionary<Vector2Int, Bitmap> communicationZones)
```

Comment:
Nitpick: This is called in the constructor so it should be static as class properties might not be set yet